### PR TITLE
enable wildcard subscribers

### PR DESCRIPTION
**Pull request by firatoltulu**
Hi,

I added a wildcard using that I think would be useful.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,4 +9,5 @@ jquery.pubsub.js
 .nyc_output/
 coverage
 # yarn lock file
-yarn.lock
\ No newline at end of file
+yarn.lock
+.vscode
```
Comment (mroderick) [inline .gitignore]: The local .gitignore is for files produces by code in the repository.
Neither of these are that.

I'd recommend putting the `.vscode` entry in your [global `.gitignore`](http://egorsmirnov.me/2015/05/04/global-gitignore-file.html) and deleting the other one.

File: src/pubsub.js [hunk 2]
```diff
@@ -69,6 +69,22 @@
...
+    function findWildCardMessage(topic)
+    {
+        var foundMessages = Object.keys(messages).find(function(messageItem){
+            return messageItem.indexOf('*') > -1 && matchMessage(topic,messageItem);
+        });
+        return foundMessages || []; 
+    }
+
+    function matchMessage(str, rule) {
+        var escapeRegex = (str) => str.replace(/([.*+?^=!:${}()|\[\]\/\\])/g, "\\$1");
```
Comment (mroderick) [inline src/pubsub.js]: This codebase should by syntactically compatible with ES3, so I am afraid that means no arrow functions.

> ES3 compatible. PubSubJS should be able to run everywhere that can execute JavaScript. Browsers, servers, ebook readers, old phones, game consoles.

It does state this in the `README.md`, but not yet in `CONTRIBUTING.md`.

File: src/pubsub.js [hunk 3]
```diff
@@ -69,6 +69,22 @@
...
+        rule = rule.split("*").map(escapeRegex).join(".*");
```
Comment (mroderick) [inline src/pubsub.js]: `.map` on Array was introduced in ES5, so you'll need to find another solution for this too.

File: src/pubsub.js [hunk 4]
```diff
@@ -69,6 +69,22 @@
         subscriber( message, data );
     }
 
+    function findWildCardMessage(topic)
+    {
+        var foundMessages = Object.keys(messages).find(function(messageItem){
```
Comment (mroderick) [inline src/pubsub.js]: `Object.keys` was introduced in ES5.1, this project is meant to be ES3 compatible.
There are other ways of finding a specific message.

File: src/pubsub.js [hunk 5]
```diff
@@ -90,6 +106,7 @@
             var topic = String( message ),
                 position = topic.lastIndexOf( '.' );
 
+            //"firat.test.1"
```
Comment (mroderick) [inline src/pubsub.js]: What does this comment mean?

File: src/pubsub.js [hunk 6]
```diff
@@ -99,6 +116,12 @@
                 position = topic.lastIndexOf('.');
                 deliverMessage( message, topic, data, immediateExceptions );
             }
+
+            
+            findWildCardMessage(message).forEach(element => {
```
Comment (mroderick) [inline src/pubsub.js]: `.forEach` is from ES5, the `=>` is from ES6.
Please use ES3

File: src/pubsub.js [hunk 7]
```diff
@@ -113,6 +136,9 @@
             found = Boolean(messages.hasOwnProperty( topic ) && hasKeys(messages[topic]));
         }
 
+        if(!found)
```
Comment (mroderick) [inline src/pubsub.js]: Please use the same brace style as the rest of the codebase. I'll try to update the eslint config to give better guidance.

File: src/pubsub.js (hunk 5 above)
Comment (firatoltulu) [reply src/pubsub.js]: remove this line, sorry :)

**Comment by firatoltulu** [comment]
Hi mroderick,

Can you merge this ? i committed last your change request

**Pull request closed without merging**